kvbc/src/categorization/kv_blockchain.cpp

```diff
@@ -103,20 +102,27 @@ BlockId KeyValueBlockchain::addBlock(CategoryInput&& category_updates,
   new_block.data.parent_digest = parent_digest_future.get();
   last_raw_block.data.parent_digest = new_block.data.parent_digest;
   block_chain_.addBlock(new_block, write_batch);
+  LOG_DEBUG(CAT_BLOCK_LOG, "Writing block [" << new_block.id() << "] to the blocks cf");
   write_batch.put(detail::BLOCKS_CF, Block::generateKey(new_block.id()), Block::serialize(new_block));
   return new_block.id();
 }
 
 std::future<BlockDigest> KeyValueBlockchain::computeParentBlockDigest(const BlockId block_id,
```

Comment:
I think now's a good time to unify and change the parent block digest calculation from CryptoPP SHA2 to OpenSSL SHA3 (using our EVP wrapper).

Reply:
We might also use ThreadPool::async instead of std::async in order to avoid thread creation on every block.

Reply:
wasn't aware of the ThreadPool class. great job btw @dartdart26

Reply:
Regarding the digest, I'll do it as part of the next PR since I have some question on it